src/workload.rs

```diff
@@ -305,10 +306,12 @@ impl WorkloadManager {
         cert_manager: Arc<SecretManager>,
     ) -> anyhow::Result<WorkloadManager> {
         let (tx, mut rx) = mpsc::channel::<Identity>(256);
-        // todo ratelimit prefetching to a reasonable limit
         tokio::spawn(async move {
             while let Some(workload_identity) = rx.recv().await {
-                match cert_manager.fetch_certificate(&workload_identity).await {
```

Comment:
nit: In case of misusing this function, maybe we should add some docs about when should use realtime fetching